app/controllers/public/addresses_controller.rb

```diff
@@ -1,16 +1,38 @@
 class Public::AddressesController < ApplicationController
   def index
+    @address = Address.new
+    @addresses = Address.where(customer: current_customer)
   end
 
   def create
+    @address = Address.new(address_params)
+    @address.customer = current_customer
+    if @address.save
+      redirect_to addresses_path
+    else
+      render :index
```

Comment:
`render`を使う場合、`index`アクションの処理を通さないので`@addresses`の指定が必要になります。
```suggestion
      @addresses = Address.where(customer: current_customer)
      render :index
```